tests/resourcecollection_test.go

```diff
@@ -0,0 +1,499 @@
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/testutils/centralgrpc"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8scorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/utils/pointer"
+)
+
+type label struct {
+	key   string
+	value string
+}
+
+var (
+	collectionNamespaces = []string{
+		"collections-test-1",
+		"collections-test-2",
+	}
+	testDeployments = []*appsv1.Deployment{
+		getTestDeployment("deployment1", []label{
+			{"key1", "value1"},
+		}...),
+		getTestDeployment("deployment2", []label{
+			{"key1", "value1"},
+			{"key2", "value2"},
+		}...),
+		getTestDeployment("deployment3", []label{
+			{"key2", "value2"},
+		}...),
+	}
+)
+
+func TestCollectionE2E(t *testing.T) {
+	suite.Run(t, new(CollectionE2ETestSuite))
+}
+
+type CollectionE2ETestSuite struct {
+	suite.Suite
+
+	ctx        context.Context
+	nsClient   k8scorev1.NamespaceInterface
+	service    v1.CollectionServiceClient
+	depService v1.DeploymentServiceClient
+}
+
+func (s *CollectionE2ETestSuite) SetupSuite() {
+	if !env.PostgresDatastoreEnabled.BooleanSetting() || !features.ObjectCollections.Enabled() {
```

Comment:
I think it's safe to make the assumption that if collections is enabled, postgres is enabled

Reply:
Changed to reflect that